perlPackages.IOPty: upgrade to 1.20

pks-t opened the pull request:
This version fixes some broken tests that started to happen in the unstable branch of nixpkgs.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

stigtsp commented:
Hm, just noticing that `IOTty` and `IOPty` points to the same tarball. Would you mind also updating this? `IOPty` should then be an alias to `IOTty` I think.

pks-t commented:
True. Adapted the PR accordingly.

pks-t commented:
Hm, doesn't seem to work as-is though. Let me have another look.

pks-t committed:
global: use IOTty instead of IOPty

These are both equivalent to one another, and we're about to convert the
IOPty package to become an alias of IOTty. It follows that the IOPty
package will only be available when `conf.allowAliases = true`.

stigtsp commented:
It also seems to trigger a mass rebuild, so would need to go into `staging` if that is the case

pks-t commented:
I've changed it to `staging` for now. Happy to revert back to `master` in case it's not needed after all.

qbit commented on pkgs/by-name/re/rex/package.nix:
```diff
@@ -70,7 +70,7 @@ perlPackages.buildPerlPackage rec {
     FileLibMagic
     HashMerge
     HTTPMessage
-    IOPty
```
Not sure if this should be changed. The rex source points to IO::Pty: https://github.com/RexOps/Rex/blob/master/t/os_dependencies.t#L14

stigtsp replied:
`IO::Pty` is included in the `IO-Tty` package so this seems fine?

qbit replied:
mk :D

pks-t committed:
perlPackages.IOTty: upgrade to 1.20

Drop the now-unneeded patch. It is not required anymore starting with
1735a78 (Make function checks more robust within shared libs,
2023-04-04), and with the patch a couple of tests fail.

pks-t committed:
perlPackages.IOPty: make it an alias of IOTty

The IOPty and IOTty packages refer to the same tarball. Make the former
an alias of the latter so that we don't have to maintain this package
twice.

pks-t commented:
@qbit @stigtsp I've rebased this pull request just now. Could you please have another look and maybe merge it? Thanks! :pray:

qbit commented:
`rex` seems happy with it!

de11n reviewed (commented):
IOTty's README says

> IO::Tty is used internally by IO::Pty to create a pseudo-tty. You wouldn't want to use it directly except to import constants, use IO::Pty.

Why are we getting rid of IO:Pty completely?

pks-t commented:
> Why are we getting rid of IO:Pty completely?

We aren't getting rid of it -- we merely make it an alias of the IOTty package given that they are both the same usptream source.

pks-t commented:
Rebased on top of latest `staging` branch. Can this be merged given that the changes got multipe approvals?

deftdawg commented:
Soooo close... just needs to hit nixpkgs-unstable and I can bring [cope to Darwin](https://github.com/NixOS/nixpkgs/pull/414717) :smile: :rocket:
